src/components/periodic-report/periodic-report.service.ts

```diff
@@ -152,13 +157,42 @@ export class PeriodicReportService {
     };
   }
 
+  async getCurrentReport(
+    parentId: ID,
+    reportType: ReportType,
+    session: Session
+  ): Promise<PeriodicReport | undefined> {
+    let interval: DurationUnit = 'quarter';
+    if (reportType === ReportType.Financial) {
+      const project = await this.project.readOne(parentId, session);
+      if (project.financialReportPeriod.value === ReportPeriod.Monthly) {
+        interval = 'month';
+      }
+    }
+
+    const currentReportStartDate = CalendarDate.local()
+      .minus({ [interval]: 1 })
+      .startOf(interval);
```

Comment:
Instead of needing to figure this out can't we just add a condition to the db query? `start < current < end`